tests/phpunit/CRM/Core/BAO/MessageTemplateTest.php

```diff
@@ -58,4 +57,323 @@ public function testCaseActivityCopyTemplate() {
+email_greeting_id:Dear {contact.first_name}
+postal_greeting_id:Dear {contact.first_name}
+addressee_id:{contact.individual_prefix}{ } {contact.first_name}{ }{contact.middle_name}{ }{contact.last_name}{ }{contact.individual_suffix}
+job_title:Busy person
+gender:Female
+birth_date:December 31st, 1998
+current_employer_id:3
+contact_is_deleted:
+created_date:January 1st, 2020 12:00 AM
+modified_date:' . CRM_Utils_Date::customFormat($modifiedDate) . '
+addressee:Mr. Robert Frank Smith II
+email_greeting:Dear Robert
+postal_greeting:Dear Robert
+current_employer:Unit Test Organization
+location_type:Home
+address_id:2
+street_address:Street Address
+street_number:123
+street_number_suffix:S
+street_name:Main St
+street_unit:45B
+supplemental_address_1:Round the corner
+supplemental_address_2:Up the road
+supplemental_address_3:By the big tree
+city:New York
+postal_code_suffix:4578
+postal_code:90210
+geo_code_1:48.858093
+geo_code_2:2.294694
+manual_geo_code:1
+address_name:The white house
+master_id:3
+county:
+state_province:TX
+country:United States
+phone:123-456
+phone_ext:77
+phone_type_id:
+phone_type:Mobile
+email:anthony_anderson@civicrm.org
+on_hold:
+signature_text:Yours sincerely
+signature_html:&lt;p&gt;Yours&lt;/p&gt;
+im_provider:1
+im:IM Screen Name
+openid:OpenID
+world_region:America South, Central, North and Caribbean
+url:http://civicrm.org
+custom_1:Mr. Spider Man II
+checksum:cs=' . $checksum . '
+contact_id:' . $tokenData['contact_id'] . '
+';
+    $this->assertEquals($expected, $messageContent['html']);
+    $this->assertEquals($expected, $messageContent['text']);
+    $this->assertEquals(str_replace("\n", '', $expected), $messageContent['subject']);
+  }
+
+  /**
+   * Gets the values needed to render domain tokens.
+   *
+   * This is keyed by all the available tokens and fills
+   * them with sample data.
+   *
+   * @return array
+   */
+  protected function getDomainTokenData(): array {
+    return [
+      'name' => 'Default Domain Name',
+      'phone' => 123,
+      'address' => [
+        'street_address' => 'Buckingham palace',
+        'supplemental_address_1' => 'Up the road',
+        'postal_code' => 90210,
+        'geocode_1' => 789,
+        'geocode_2' => 890,
+        'city' => 'London',
+      ],
+      'email' => 'crown@example.com',
+      'id' => CRM_Core_Config::domainID(),
+      'description' => 'rather nice',
+    ];
+  }
+
+  /**
+   * Get all the available tokens with usable data.
+   *
+   * This is the result rendered from CRM_Core_SelectValues::contactTokens();
+   * and has been gathered by calling that function.
+   *
+   * I have hard-coded them so we have a record of what we have
+   * seemingly committed to support.
+   *
+   * Note it will render additional custom fields if they exist.
+   *
+   * @return array
+   * @throws \API_Exception
+   * @throws \CRM_Core_Exception
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function getAllContactTokens(): array {
+    return [
```

Comment:
@totten this & the function below (for domains) provide a full set of token data for the relevant entity. You might be able to extract to a json sample but there are a couple of gotchas around the rows that reference related entities

You can see how each token presents in https://github.com/civicrm/civicrm-core/pull/19551/files#diff-c3c60600f080b49f2a9029b82865098c2a2742fbcedce0ecd731376917d80e1bR162

Note this PR should be merged & I'll rebase #19550 because I think we will need to discuss the slight differences #19550 highlights in the test run & this actually fixes an error in the extraction (& adds tests)